Assignment6.txt

```diff
@@ -0,0 +1,37 @@
+Madeleine Charity 
+
+1. 
+    Is there anything wrong with this code? If so, what is it and how would you fix it?
+
+    #include <stdio.h> 
+    int main() {
+    int num, n;
+    int arr[num];
+    printf(“Enter the size of an array”);
+    scanf(“%d”, &num);
+    for (n=0; n<num; n++) {
+    	arr[n] = 0;
+    }		 
+    }
+    
+    yes, the printf and scanf staments should be before the array declarion and for loop. 
+
+    #include <stdio.h> 
+    int main() {
+    int num, n;
+    printf(“Enter the size of an array”);
+    scanf(“%d”, &num);
+    int arr[num];
+    for (n=0; n<num; n++) {
+    	arr[n] = 0;
+    }		 
+    }
+
+    2. A segmention fault is when the program tries to access memory that it isn't allowed to. For example, if you tried to access arr[1000] if arr was size 10. 
```

Comment:
right! as far as the example, that's not necessarily considered a segmentation fault. It _could_ result in one, but C actually defines array access errors as _undefined_ behavior